src/snowflake/snowpark/types.py

```diff
@@ -41,6 +42,23 @@ def __repr__(self):
     def is_primitive(self):
         return True
 
+    @classmethod
+    def type_name(cls) -> str:
+        return cls.__name__[:-4].lower()
+
+    def simple_string(self) -> str:
+        return self.type_name()
+
+    def json_value(self) -> Union[str, Dict[str, Any]]:
+        return self.type_name()
+
+    def json(self) -> str:
+        return json.dumps(self.json_value(), separators=(",", ":"), sort_keys=True)
+
+    typeName = type_name
```

Comment:
Does pyspark have typeName only, or also type_name? If typeName only, I don't think we need to add type_name

Reply:
they only have typeName, but type_name is a python style naming that we should apply to our code right? We just want to provide compatibility with pyspark

Reply:
Then I don't think we need to add type_name and other methods that pyspark doesn't have - we only need to add APIs that pyspark has. cc @sfc-gh-aalam

Reply:
snowpark way is to follow snake_case and we create alias for pyspark counterpart for better migration. This is what I think we should keep following along

https://github.com/snowflakedb/snowpark-python/blob/8b6090ab2e2fde1408afa38693b3d223970890d7/src/snowflake/snowpark/dataframe.py#L4317-L4345